src/Symfony/Component/TypeInfo/Type/UnionType.php

```diff
@@ -100,7 +99,10 @@ public function __toString(): string
         $string = '';
         $glue = '';
 
-        foreach ($this->types as $t) {
+        $types = $this->types;
+        usort($types, fn (Type $a, Type $b): int => (string) $a <=> (string) $b);
+
+        foreach ($types as $t) {
             $string .= $glue.($t instanceof CompositeTypeInterface ? '('.$t.')' : $t);
             $glue = '|';
         }
```

Comment:
We can make this a bit more performant by converting all types to string before sorting; avoiding multiple calls to `toString`.

```suggestion
        $types = array_map(fn (Type $type): string => ($t instanceof CompositeTypeInterface ? '('.$t.')' : $t), $this->types);
        sort($types);
        $string = implode('|', $types);
```

The only difference is that composite types will be first.

But are you sure the types should be sorted in the string representation? If we decide to not sort them in `getTypes()`. All representations should be consistent.

Reply:
You're right, after a second thought, sorting types can be avoided!